transactionoutbox-core/src/test/java/com/gruelbox/transactionoutbox/TestDefaultPersistorConfiguration.java

```diff
@@ -35,6 +38,30 @@ final void whenMigrateIsFalseDoNotMigrate() throws Exception {
         });
   }
 
+  @Test
+  final void writeSchema() {
+    StringWriter stringWriter = new StringWriter();
+
+    DefaultPersistor defaultPersistor = DefaultPersistor.builder().dialect(Dialect.H2).build();
```

Comment:
You'll get a NPE when calling `writeSchema` if you don't supply a dialect here. The validation does not happen until `TransactionOutboxImpl` is constructed.